app/grandchallenge/reader_studies/tasks.py

```diff
@@ -90,6 +92,62 @@ def add_image_to_display_set(
             display_set.values.add(civ)
 
 
+@shared_task(**settings.CELERY_TASK_DECORATOR_KWARGS["acks-late-2xlarge"])
+def add_file_to_display_set(
+    *,
+    user_upload_pk,
+    display_set_pk,
+    interface_pk,
+    civ_pk=None,
+):
+    user_upload = UserUpload.objects.get(pk=user_upload_pk)
+    display_set = DisplaySet.objects.get(pk=display_set_pk)
+    interface = ComponentInterface.objects.get(pk=interface_pk)
+    with transaction.atomic():
+        if civ_pk is None:
+            civ = ComponentInterfaceValue.objects.create(interface=interface)
+        else:
+            civ = ComponentInterfaceValue.objects.get(pk=civ_pk)
```

Comment:
Changing the value of the CIV should never be done. A new CIV should always be created (unless the file, value and image is already none, but then it is strange to create the CIV without a value elsewhere first).

Check the usage of the form too, and other places where you may do this.